nucleus/job.py

```diff
@@ -0,0 +1,52 @@
+from dataclasses import dataclass
+import time
+from typing import Dict, List
+
+import requests
+
+
+@dataclass
+class AsyncJob:
+    id: str
+    client: "NucleusClient"  # type: ignore # noqa: F821
+
+    def status(self) -> Dict[str, str]:
+        return self.client.make_request(
+            payload={},
+            route=f"job/{self.id}",
+            requests_command=requests.get,
+        )
+
+    def errors(self) -> List[str]:
+        return self.client.make_request(
+            payload={},
+            route=f"job/{self.id}/errors",
+            requests_command=requests.get,
+        )
+
+    def sleep_until_complete(self, verbose_std_out=True):
```

Comment:
It's also possible here to loop forever if somehow the step function doesn't correctly record status errored.  Could add a max number of loops if you feel it's appropriate

Reply:
It prints once a second which I like personally. If we want it less often, I can just tweak this parameter.

Looping forever is the expected behavior on the client side in that case since the job is showing as running. In an interactive ipython notebook, it's easy to stop execution.